### DISCUSSION TITLE
MerchantFulfillment  code is error

### DISCUSSION THREAD

**Discussion by herolzwcj** [abuzuhri/Amazon-SP-API-CSharp#45] [General]
During the operation of the merchant fulfillment interface, I found that the contents of the files in the merchant fulfillment models folder were wrong

**Comment by abuzuhri** [comment]
> During the operation of the merchant fulfillment interface, I found that the contents of the files in the merchant fulfillment models folder were wrong



Can you please give more details more description so I will double check this

**Reply by herolzwcj** [reply]
ex. When I perform the following operations, I find that I cannot initialize the parameters。 tks!!

 private async void GetAmz(AmazonStoreInfo amazonStoreInfo) {
            var amazonConnection = await GetAmazonConnection(amazonStoreInfo);
            amazonConnection.MerchantFulfillment.GetEligibleShipmentServices(new GetEligibleShipmentServicesRequest { 
             ShipmentRequestDetails = new ShipmentRequestDetails { 
              AmazonOrderId=(new AmazonOrderId())    **// AmazonOrderId ? This parameter should not be a string type**
             },
             ShippingOfferingFilter = new ShippingOfferingFilter { 
              CarrierWillPickUp = CarrierWillPickUpOption.ShipperWillDropOff,
               IncludeComplexShippingOptions = false,
                DeliveryExperience = DeliveryExperienceOption.NoTracking,
              IncludePackingSlipWithLabel = false
             }
            });

**Reply by abuzuhri** [reply]
> AmazonOrderId=(new AmazonOrderId()) **// AmazonOrderId ? This parameter should not be a string type**

You mean this should be string ?

**Reply by herolzwcj** [reply]
I found that many of the classes in the following directory can not match with the manual of AMZ API, 
so I failed to initialize to request the parameters. Can you correct it? tks!!!!

**Amazon-SP-API-CSharp/Source/FikaAmazonAPI/AmazonSpApiSDK/Models/MerchantFulfillment/**


onConnection.MerchantFulfillment.GetEligibleShipmentServices(new GetEligibleShipmentServicesRequest
            {
                ShipmentRequestDetails = new ShipmentRequestDetails
                {
                    AmazonOrderId = aorder.AmazonOrderId,
                    Weight = new Weight
                    {
                        Unit = UnitOfWeight.Oz,
                        Value = (PackageDimension)0.1
                    },
                    PackageDimensions = new PackageDimensions
                    {
                        **Height =(PackageDimension) 0.1, //? ex.
                        Length = (PackageDimension)0.1,//? ex,
                        Width = (PackageDimension)0.1,//? ex.**
                        Unit = UnitOfLength.Inches
                    },
                    ShippingServiceOptions = new ShippingServiceOptions
                    {
                        CarrierWillPickUp = false,
                        DeclaredValue = new CurrencyAmount { 
                         Amount= 0.1,
                         CurrencyCode = "USD"

**Reply by abuzuhri** [reply]
PackageDimension fixed in v1.0.24

This class generated by swagger hopefully no more problem

**Comment by abuzuhri** [comment]
Can you please check v1.0.23 , i fix that bug

**Reply by herolzwcj** [reply]
tks！

**Reply by abuzuhri** [reply]
more fix in v1.0.24
